src/team-calendar-service.ts

```diff
@@ -26,20 +25,27 @@ export function serviceHandler(srv) {
       });
   });
 
-  srv.after("UPDATE", "Appointment", async ([appointment]: [Appointment]) => {
-    // split into multiple days and write in parallel
-    const person = await personById(srv, appointment.person_ID);
-
-    return Promise.all(
-      splitAppointmentIntoDays(appointment)
-        .map(day => buildTimeSheetEntry(appointment, person, day))
-        .map(writeTimeSheetEntry)
-    )
-    .then(() => {
-      return appointment;
-    })
-    .catch(error => {
-      throw Error(`Failed to create appointment! ${error.message}`);
-    });
+  srv.after("UPDATE", "Appointment", async (appointment: Appointment, req) => {
+    //the transaction joins the previous read request
+    const tx = srv.transaction(req);
+    const persons = await tx.run(SELECT.from ("Person").where({ID:appointment.person_ID}));
+    if (persons.length === 0) {
+      throw new Error(`No person found with ID ${appointment.person_ID}`);
+    }
+    else{
```

Comment:
You don't need the else here, because either you have found a person or an error was already thrown.